packages/slate-editor/src/components/OptionsGroup/OptionsGroup.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+
+import type { OptionsGroupOption, ChooseOptionProps } from './Option';
+import { ChooseOption } from './Option';
+import styles from './OptionsGroup.module.scss';
+
+interface OptionsGroupCommon<T extends string> {
+    name: string;
+    options: OptionsGroupOption<T>[];
+    columns?: number;
+}
+
+type OptionsGroupProps<T extends string> = OptionsGroupCommon<T> &
+    Omit<ChooseOptionProps<T>, 'option'>;
```

Comment:
nit: I find it difficult to trace that `OptionsGroupProps` extends `OptionsGroupCommon` and `ChooseOptionProps`. Let's just inline those props here. I see no harm in slight duplication here.